runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -12229,6 +12229,447 @@ static bool inlineIntrinsicInflate(TR::Node *node, TR::CodeGenerator *cg)
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, vectorLoopLabel);
+   // Load 16 items; for p9+ we can count zeroes in a vector register, so loading in order matters;
+   // before p9, we can't load bytes in the correct order anyways, so we ignore the ordering here.
+   if (p9Plus)
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::lxvb16x, node, vtmp1Reg, startReg, indexReg);
+   else
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::lxvw4x, node, vtmp1Reg, startReg, indexReg);
+   // bit 2 of cr6 (ZERO) will not be set if any comparison is true
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::vcmpgtsb_r, node, vtmp1Reg, vconstant0Reg, vtmp1Reg);
+   // branch when the ZERO bit is not set
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::bne, node, matchLabel, cr6);
+
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, indexReg, indexReg, 16);
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::cmp4, node, cr0, indexReg, tempReg);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::blt, node, vectorLoopLabel, cr0);
+
+   // go to residue if we don't have enough items to do one load
+   // We kinda have to tuck the code checking which residual handler to use here:
+   //    no item - end; 1 item - byte; 2-3 items - hword; 4-15 items - word/dword.
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::subf_r, node, tempReg, indexReg, lengthReg);
+   if (isCountPositives)
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::beq, node, endLabel, cr0);
+   else
+      generateConditionalBranchInstruction(cg, TR::InstOpCode::beq, node, noMatchLabel, cr0);
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::cmpi4, node, cr0, tempReg, 4);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::bge, node, serialDWordCheckLabel, cr0);
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::cmpi4, node, cr0, tempReg, 2);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::bge, node, serialHWordLabel, cr0);
+   generateLabelInstruction(cg, TR::InstOpCode::b, node, serial1Label);
+
+   // --- when there is a match but we don't know the exact location yet
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, matchLabel);
+   if (isCountPositives)
+      {
+      if (p9Plus) // just count for P9+
+         {
+         generateTrg1Src1Instruction(cg, TR::InstOpCode::vclzlsbb, node, tempReg, vtmp1Reg);
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::add, node, indexReg, tempReg, indexReg);
+         generateLabelInstruction(cg, TR::InstOpCode::b, node, endLabel);
+         }
+      else // otherwise, we simply go through the last 16 items as if they are the last 16
+         {
+         // we are not worried about overwriting length, since we must be able to find a match
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, lengthReg, indexReg, 15);
```

Comment:
tempReg is going to be 15 definitely. why `subf `to get the 15? couldn't you `li` 15 into it?